htlcswitch/link.go

```diff
@@ -448,6 +450,10 @@ func NewChannelLink(cfg ChannelLinkConfig,
 	channel *lnwallet.LightningChannel) ChannelLink {
 
 	logPrefix := fmt.Sprintf("ChannelLink(%v):", channel.ChannelPoint())
+	quitCtx, quitFunc := context.WithCancel(context.Background())
+
+	// Initialize the Done channel for our quit context.
+	_ = quitCtx.Done()
```

Comment:
TIL it needs to be initialized: https://cs.opensource.google/go/go/+/refs/tags/go1.23.2:src/context/context.go;l=438-451

Reply:
I think it's doesn't really have to be, but if we wanted to save that channel and then use it later (call `Done()` once) we'd want to explicitly initialize it as here.